frontend/lib/resolution/resolution-queries.cpp

```diff
@@ -3625,6 +3625,29 @@ doIsCandidateApplicableInitial(ResolutionContext* rc,
     }
   }
 
+  // For now, ignore any candidates that are not part of the 2.0 edition
+  // Going forward, we'll need to teach Dyno's resolver to respect the
+  // --edition flag.
+  //
+  // Developed by @benharsh
+  if (!candidateId.isEmpty()) {
+    if (auto ast = parsing::idToAst(context, candidateId);
+        ast && ast->isFunction()) {
+      auto fn = ast->toFunction();
+      if (auto attr = fn->attributeGroup();
+          attr && attr->hasEdition()) {
+        auto first = attr->firstEdition();
+        auto last = attr->lastEdition();
+        auto two = UniqueString::get(context, "2.0");
```

Comment:
This is probably going to be written more and more often as more "next edition" things are added. I'd recommend adding the strings `"2.0"`, `"default"`, and `"pre-edition"` to the header `frontend/include/chpl/framework/all-global-strings.h`. Then you can simply use the shorthand `USTR("2.0")` or `USTR("pre-edition")`.

Reply:
I think I'm going to leave this as-is tonight since I'm not a Dyno implementer and got it from @benharsh (so don't want to take sides between the two of you), but I'm open to refactoring this more post-merge if he agrees.  Note that I imagine we'll likely be revisiting or replacing this code as we teach Dyno how to properly resolve editions.